### PR TITLE
EPMDJ-5260: Add the ability to toggle baseline

### PR DESCRIPTION
**Pull request by RomanAchilov**
Issue: EPMDJ-5260

### CODE REVIEW

File: src/pages/agent-full-page/coverage-plugin/api/toggle-baseline.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  pluginId: string,
+  { onSuccess, onError }: { onSuccess?: () => void; onError?: (message: string) => void } = {},
+): Promise<void> {
+  try {
+    await axios.post(`/agents/${agentId}/plugins/${pluginId}/dispatch-action`, {
+      type: 'TOGGLE_BASELINE',
+    });
+    onSuccess && onSuccess();
+  } catch ({ response: { data: { message } = {} } = {} }) {
+    onError && onError(message);
```
Comment (pavyarov) [inline src/pages/agent-full-page/coverage-plugin/api/toggle-baseline.ts:14]: No default message?

File: src/pages/agent-full-page/coverage-plugin/coverage-plugin-header/coverage-plugin-header.tsx [hunk 2]
```diff
@@ -51,18 +55,66 @@ export const CoveragePluginHeader = coveragePluginHeader(({ className }: Props)
     metrics: { coverage, risks: risksCount, tests: testToRunCount },
   };
   const StatusIcon = Icons[status];
+  const isActiveBuild = activeBuildVersion === buildVersion;
+
+  const { version: baseline } = useBuildVersion<Record<string, string>>('/data/baseline') || {};
```
Comment (pavyarov) [inline src/pages/agent-full-page/coverage-plugin/coverage-plugin-header/coverage-plugin-header.tsx:60]: Record?

File: src/pages/agent-full-page/coverage-plugin/coverage-plugin-header/coverage-plugin-header.tsx [hunk 3]
```diff
@@ -51,18 +55,66 @@ export const CoveragePluginHeader = coveragePluginHeader(({ className }: Props)
     metrics: { coverage, risks: risksCount, tests: testToRunCount },
   };
   const StatusIcon = Icons[status];
+  const isActiveBuild = activeBuildVersion === buildVersion;
+
+  const { version: baseline } = useBuildVersion<Record<string, string>>('/data/baseline') || {};
+  const { version: previousBuildVersion = '' } = useBuildVersion<ParentBuild>('/data/parent') || {};
+  const [isBaseline, setIsBaseline] = React.useState(false);
+  React.useEffect(() => {
+    baseline && setIsBaseline(baseline === buildVersion);
```
Comment (pavyarov) [inline src/pages/agent-full-page/coverage-plugin/coverage-plugin-header/coverage-plugin-header.tsx:64]: Funny

**Review by pavyarov — CHANGES_REQUESTED** [review]

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

File: src/pages/agent-full-page/coverage-plugin/api/toggle-baseline.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  agentId: string,
+  pluginId: string,
+  { onSuccess, onError }: { onSuccess?: () => void; onError?: (message: string) => void } = {},
+): Promise<void> {
+  try {
+    await axios.post(`/agents/${agentId}/plugins/${pluginId}/dispatch-action`, {
+      type: 'TOGGLE_BASELINE',
+    });
+    onSuccess && onSuccess();
+  } catch ({ response: { data: { message } = {} } = {} }) {
```
Comment (RomanDavlyatshin) [inline src/pages/agent-full-page/coverage-plugin/api/toggle-baseline.ts:13]: `data` might be undefined, that's gonna break

File: src/pages/agent-full-page/coverage-plugin/coverage-plugin-header/coverage-plugin-header.tsx (hunk 3 above)
Comment (RomanDavlyatshin) [reply src/pages/agent-full-page/coverage-plugin/coverage-plugin-header/coverage-plugin-header.tsx:64]: Can you elaborate, please? 😃

File: src/pages/agent-full-page/coverage-plugin/coverage-plugin-header/coverage-plugin-header.tsx [hunk 5]
```diff
@@ -52,20 +67,74 @@ export const CoveragePluginHeader = coveragePluginHeader(({ className }: Props)
...
-        <CurrentBuild>
-          Build:
-          <Version data-test="coverage-plugin-header:build-version">{buildVersion}</Version>
-        </CurrentBuild>
+        <Panel>
+          <CurrentBuild>
+            Build:
+            <Version data-test="coverage-plugin-header:build-version">{buildVersion}</Version>
+          </CurrentBuild>
+          {((!previousBuildVersion && isBaseline) ||
```
Comment (RomanDavlyatshin) [inline src/pages/agent-full-page/coverage-plugin/coverage-plugin-header/coverage-plugin-header.tsx:84]: Refactor repeated conditions checks (1. before &&, 2. `disabled` prop, 3. inside Panel) to a single function call

File: src/pages/agent-full-page/coverage-plugin/coverage-plugin-header/coverage-plugin-header.tsx [hunk 6]
```diff
@@ -90,45 +159,61 @@ export const CoveragePluginHeader = coveragePluginHeader(({ className }: Props)
...
           onClick={() => push(`/full-page/${agentId}/${buildVersion}/${pluginId}/tests-to-run`)}
         />
       </Actions>
-      {isRisksModalOpen && (
-        <RisksModal
-          isOpen={isRisksModalOpen}
-          onToggle={setIsRisksModalOpen}
-        />
-      )}
+      {isRisksModalOpen && <RisksModal isOpen={isRisksModalOpen} onToggle={setIsRisksModalOpen} />}
```
Comment (RomanDavlyatshin) [inline src/pages/agent-full-page/coverage-plugin/coverage-plugin-header/coverage-plugin-header.tsx:179]: I don't understand: what's the point in passing `isRisksModalOpen` to a prop `isOpen` along with `{isRisksModalOpen && (` check?
I mean, if `isRisksModalOpen === false` then `<RisksModal>` won't even render, right?
Either do the check (and remove `isOpen`) or remove the check and let `<RisksModal>` to manage its rendering with `isOpen`.

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

**Review by RomanDavlyatshin — APPROVED** [review]

**Pull request merged**